### DISCUSSION TITLE
CSS not applying

### DISCUSSION THREAD

**Discussion by jordanhandy** [retypeapp/retype#222] [Q&A] [answered]
Hello,

The CSS styles for my pages don't appear to be applying.
What am I missing?  I ensured that my `url` defined in the .yml file was correct (I tried both with a trailing "/" and without, on the URL.

The page is here, [Page Link](https://jordanhandy.github.io/obsidian-cloudinary-uploader/) and the [Repo Link](https://github.com/jordanhandy/obsidian-cloudinary-uploader).

I followed the instructions in the Retype docs for Github Actions, but I may be missing a minor detail?

Thanks

**Comment by jordanhandy** [comment, marked as answer]
I ended up fixing this.  In case someone runs across it.  

For some reason, Retype didn't like that I had the yml config file in a subdirectory, even though my input key value was correct relative to its location.  I moved the retype yml config file to the root of my project and updated the input path again relative to root.  All worked

**Reply by geoffreymcgill** [reply]
I'm glad to see your site is working correctly, although...

> For some reason, Retype didn't like that I had the yml config file in a subdirectory

I took a quick look through your repo history, but it was not immediately clear to me why your original `retype.yml` configuration was not generating the correct resource paths. We will replay your repo history and try to figure out what was going wrong. 

The `retype.yml` file does not need to be the root of the repo, although sometimes getting the combination of [`input`](https://retype.com/configuration/project/#input) and [`url`](https://retype.com/configuration/project/#url) configured correctly can require some fiddling.

If the `.js` or `.css` files are not loading, you can view the HTML source for that page and take a look at the paths to those files. Typically, an incorrectly configured `url` would be the cause. Either an extra folder was added to the `url` or is missing. 

If you are publishing using GitHub Pages, that URL and path may be different than what is configured in the `retype.yml` `url` config.

**Reply by geoffreymcgill** [reply]
> You might have a hard time looking through the recent commit history as I rebased to get rid of some of the garbage commits while I tested things out.

No problem. Your next comment gives me a clue as to what happened. 

> On my initial run of retype watch a brand new yml file was generated for me in the project root, even though I already had one in the subdir I wanted. I may have ran the command in the wrong path initially, I'm not sure.

Ya, I think that might be it. If your command prompt is in a different location than the `retype.yml` file, Retype will assume this is a fresh project and create a new default `retype.yml` file. 

The command prompt can be in a different location than the `retype.yml` file, but you just need to pass the path to the `retype.yml` as an argument. For instance, if you were in the root of your repo and the location of the Retype powered docs project was `docs/retype.yml`, then you would call the command `retype watch docs/retype.yml`. Even just `retype watch docs` would work in that scenario. The key is pointing to the `retype.yml` file or the directory in which it is contained.

Lots of pathing options are possible, including relative paths, such as `retype watch ../../my-project/docs/` if you need to jump up and down a few directories.

Hope this helps.

**Reply by geoffreymcgill** [reply]
If you are storing your `retype.yml` file in a location other than the root of the repo and using the Retype GitHub Action, you need to point the GitHub Action workflow to that new location. Adding the following to your `.github/workflows/retype.yml` file would be required:

```yml
- uses: retypeapp/action-build
  with:
    config: docs
```
More details available at:
https://github.com/retypeapp/action-build#specify-path-to-the-retypeyml-file

Hope this helps.

**Comment by jordanhandy** [comment]
That makes sense. I didn't have that in my yml initially.

Jordan Handy
Architect
Salesforce ¦ AWS ¦ Oracle Cloud

**Reply by fabriciomurta** [reply]
Just in case, we've logged an enhancement to the build action (retypeapp/action-build#16) so that it looks a bit further for configuration files. Instead of just "going default" in case it finds a config file one or two directories deep, it could use it. In case it finds more than one configuration file candidates though, it would not run and provide helpful messages to either remove duplicate config files or explicitly specify the path for the wanted one.

**Reply by fabriciomurta** [reply]
We have just implemented and pushed to action-build@v1 the enhancement as detailed in retypeapp/action-build#16.

in short:
- if `config` provided, just use it (retained behavior)
- if retype config found at root, use it (retained behavior)
- look for retype config files down two directory levels from the repository root
- if just one file found, use it;
- no file found, init default (like happened here originally, but shows a warning in workflow summary now)
- multiple files found, fail and show descriptive error message to workflow summary
